daml/Marketplace/Rule/AllocationAccount.daml

```diff
@@ -38,7 +39,14 @@ template AllocationAccountRule
       credit : Asset -> Update (ContractId AssetDeposit)
       credit = \asset -> create AssetDeposit with ..
       creditAccount : Account -> Asset -> Update (ContractId AssetDeposit)
-      creditAccount = \account asset -> exerciseByKey @AssetSettlementRule account.id AssetSettlement_Credit with ctrl = nominee, ..
+      creditAccount = \account asset -> do
+        depositCid <- exerciseByKey @AssetSettlementRule account.id AssetSettlement_Credit with ctrl = nominee, ..
+        deposit <- fetch depositCid
+        let
+          addProvider    = exerciseByKey @AssetSettlementRule account.id AssetSettlement_AddController with ctrl = deposit.account.owner
+          removeProvider = exerciseByKey @AssetSettlementRule account.id AssetSettlement_RemoveController with ctrl = deposit.account.owner
+          addObservers   = exercise depositCid AssetDeposit.AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
+        addProvider *> addObservers <* removeProvider
```

Comment:
this causes an error in E2E test bc it lacks authorization from Alice

.. so still workin on this part